cg/meta/workflow/mutant/mutant.py

```diff
@@ -249,3 +267,57 @@ def link_nanopore_fastq_for_sample(
         LOG.info(f"Concatenation in progress for sample {sample.internal_id}.")
         self.fastq_handler.concatenate(read_paths, concatenated_path)
         self.fastq_handler.remove_files(read_paths)
+
+    def run_qc_and_fail_analyses(self, dry_run: bool) -> None:
+        """Run qc check, report qc summaries on Trailblazer and fail analyses that fail QC."""
+        cases_qc_ready: list[Case] = self.get_completed_cases()
+        LOG.info(f"Found {len(cases_qc_ready)} cases to perform QC on!")
+
+        for case in cases_qc_ready:
+            self.trailblazer_api.set_analysis_status(
+                case_id=case.internal_id, status=AnalysisStatus.QC
+            )
+
+            try:
+                qc_result: QualityResult = self.get_qc_result(case=case)
+            except Exception:
+                LOG.error(f"Could not run QC for case {case.internal_id}")
+                self.trailblazer_api.set_analysis_status(
+                    case_id=case.internal_id, status=AnalysisStatus.ERROR
+                )
+                raise CgError(f"Could not run QC for case {case.internal_id}")
+
+            self.report_qc_on_trailblazer(case=case, qc_result=qc_result)
+
+            if not qc_result.case.passes_qc:
+                if not dry_run:
+                    self.fail_analysis(case)
+            else:
+                self.trailblazer_api.set_analysis_status(
+                    case_id=case.internal_id, status=AnalysisStatus.COMPLETED
+                )
+
```

Comment:
In line 284, the status is changed to QC when performing QC. I thought this makes logical sense, but in practise maybe it will be so fast that the QC status will barely be noticed. Iddeally the QC status would be set as soon as the analyses finish (instead of complete) but that requires bigger changes that are out of this scope. Let me know what you think of all of this!

Reply:
Will remove QC status so this does not apply anymore.